### PR TITLE
Use SBOM descriptor version

### PR DESCRIPTION
**Pull request by jonasagx**
Fix lack of version info when Syft is used a library.

| Format | Pass/Fail | Notes |
| ------- | --------- | ------ |
| syft-json | 🟢 | |
| cyclonedx-xml | 🟢 | |
| cyclonedx-json | 🟢 | |
| github-0-json | 🟢 | Reports `0.0.0-dev` |
| spdx-tag-value | 🟢 | |
| spdx-json | 🟢  | |
| table | 🟢 | N/A |
| text | 🟢 | N/A |

Fix: https://github.com/anchore/syft/issues/1010

Signed-off-by: Jonas Xavier <jonasx@anchore.com>

### CODE REVIEW

File: internal/formats/spdx22tagvalue/test-fixtures/snapshot/TestSPDXJSONSPDXIDs.golden [hunk 1]
```diff
@@ -2,11 +2,11 @@ SPDXVersion: SPDX-2.2
...
 SPDXID: SPDXRef-DOCUMENT
 DocumentName: .
-DocumentNamespace: https://anchore.com/syft/dir/8fbb3714-785d-4e3e-95cf-44a258bc65b0
-LicenseListVersion: 3.16
+DocumentNamespace: https://anchore.com/syft/dir/5c828193-2b83-4fa1-8861-25c334ba8aa7
+LicenseListVersion: 3.17
 Creator: Organization: Anchore, Inc
-Creator: Tool: syft-[not provided]
-Created: 2022-05-02T15:27:05Z
+Creator: Tool: syft-
```
Comment (wagoodman) [inline internal/formats/spdx22tagvalue/test-fixtures/snapshot/TestSPDXJSONSPDXIDs.golden:8]: should the version be empty here?

Comment (wagoodman) [inline internal/formats/spdx22tagvalue/test-fixtures/snapshot/TestSPDXJSONSPDXIDs.golden:8]: should the version be empty here?

Comment (jonasagx) [reply internal/formats/spdx22tagvalue/test-fixtures/snapshot/TestSPDXJSONSPDXIDs.golden]: nope, good catch, the original test was missing an sobs Descriptor. Fixed now.

**Review by spiffcs — APPROVED** [review]
Thanks for updating the descriptor here from the earlier comments. LGTM

**Comment by wagoodman** [comment]
didn't get to this before the merge, can we add some explicit tests to ensure we grab the version from the descriptor and not the lib approach (so not lean on snapshot tests here)?

**Pull request merged**